### PR TITLE
Add Unresponsive Slashing Examples and Edits to Staking Page

### LINKED ISSUE #2612: New Page for Slashing focusing on "Lessons Learned from [Past] Slashing"
"We should create a page "Lessons learned from past slashing events" to group everything together. Looks like the slashing conditions are still quite scattered around." -Wei Tang

Perhaps here: https://wiki.polkadot.network/docs/maintain-guides-secure-validator or in a separate page under Validator section?

### LINKED ISSUE #2936: Staking page, Slashing section issues
https://wiki.polkadot.network/docs/learn-staking#slashing

"Let us run through a few examples to understand this equation. In all of the examples, assume that there are 100 validators in the active set."

But there are no examples below that...

"The penalty would be Min((3 5 / 100)^2, 1) = 0.0225, or a 2.25% slash. If 20 validators equivocate, this is a much more serious offence and possibly indicates a coordinated attack on the network, and so the slash will be much greater - Min((3 20 / 100)^2, 1) = 0.36, or a 36% slash on all of these validators and their nominators. All slashed validators will also be chilled."

There are some extraneous italics here..

### PR DESCRIPTION
**Pull request by salmad3**
Linked issue describes next steps for the section.

Fixes #2936

### CODE REVIEW

File: docs/learn/learn-staking.md [hunk 1]
```diff
@@ -321,9 +321,9 @@ validator. The penalty would be Min(3 \* 1 / 100)^2, 1) = 0.0009, or a 0.09% sla
 validator pool (i.e., all stake held by the validator and its nominators).
 
 Now assume that there is a group running several validators, and all of them have an issue in the
-same slot. The penalty would be Min((3 _ 5 / 100)^2, 1) = 0.0225, or a 2.25% slash. If 20 validators
+same slot. The penalty would be Min((3 \_ 5 / 100)^2, 1) = 0.0225, or a 2.25% slash. If 20 validators
```
Comment (profradha) [inline docs/learn/learn-staking.md]: I don't understand the usage of underscore symbol here. Isn't it supposed to be ((3 * 5)/100)^2 which is equal to 0.0225 and min(0.0225, 1) = 0.0225

File: docs/learn/learn-staking.md [hunk 2]
```diff
@@ -321,9 +321,9 @@ validator. The penalty would be Min(3 \* 1 / 100)^2, 1) = 0.0009, or a 0.09% sla
...
 equivocate, this is a much more serious offence and possibly indicates a coordinated attack on the
-network, and so the slash will be much greater - Min((3 _ 20 / 100)^2, 1) = 0.36, or a 36% slash on
+network, and so the slash will be much greater - Min((3 \_ 20 / 100)^2, 1) = 0.36, or a 36% slash on
```
Comment (profradha) [inline docs/learn/learn-staking.md]: Same as above

File: docs/learn/learn-staking.md [hunk 3]
```diff
@@ -289,17 +289,23 @@ Here is the formula for calculation:
...
+and being slashed for equivocation - see below). A comprehensive guide on validator setup is
+available [here](../maintain/maintain-guides-secure-validator.md).
 
 Validators should have a well-architected network infrastructure to ensure the node is running to
 reduce the risk of being slashed or chilled. A high availability setup is desirable, preferably with backup
 nodes that kick in **only once the original node is verifiably offline** (to avoid double-signing
 and being slashed for equivocation - see below). A comprehensive guide on validator setup is
 available [here](../maintain/maintain-guides-secure-validator.md).
 
+Let us run through a few examples to understand this equation. In all of the examples, assume that
```
Comment (profradha) [inline docs/learn/learn-staking.md]: Which equation? Also where is the example Bill asked to add in #2936 ?

File: docs/learn/learn-staking.md (hunk 1 above)
Comment (profradha) [reply docs/learn/learn-staking.md]: Using \* instead of \_ might fix the issue and stay true to the equation

**Review by DrW3RK — CHANGES_REQUESTED** [review]
I do not see an example explaining the equation for slashing pertaining to unresponsiveness. Either remove that sentence completely or add an example below to fix #2936

File: docs/learn/learn-staking.md (hunk 1 above)
Comment (salmad3) [reply docs/learn/learn-staking.md]: Addressed.

File: docs/learn/learn-staking.md (hunk 2 above)
Comment (salmad3) [reply docs/learn/learn-staking.md]: Addressed.

File: docs/learn/learn-staking.md (hunk 3 above)
Comment (salmad3) [reply docs/learn/learn-staking.md]: Addressed.

**Comment by salmad3** [comment]
> I do not see an example explaining the equation for slashing pertaining to unresponsiveness. Either remove that sentence completely or add an example below to fix #2936

Yeah misunderstood what the content was trying to led into (thought the example was referring to the equivocation examples below). The equation for unresponsiveness also seems incorrect after looking into it further. Please also verify the example @DrW3RK

File: docs/learn/learn-staking.md [hunk 4]
```diff
@@ -287,49 +287,40 @@ Here is the formula for calculation:
...
-
-Let us run through a few examples to understand this equation. In all of the examples, assume that
-there are 100 validators in the active set.
-
-> Note that if less than 10% of all validators are offline, no penalty is enacted.
-
-### GRANDPA Equivocation
+Here is an example of unresponsiveness. If there are 100 validators in the active set, 
+and 1 validator is unresponsiveness, the unresponsiveness validator would be slashed at a fraction of 
+0.05 * min((3 * (x - 1))) / n, 1) which comes out to a 5% slash.
```
Comment (profradha) [inline docs/learn/learn-staking.md]: If one validator is unresponsive, then min(0,1) would be 0 and hence no slashing will be applied.

I think a deeper understanding of Slashing mechanisms backed by the code implementation should be used before writing examples for this. Please take some time in drafting the examples.

**Review by DrW3RK — CHANGES_REQUESTED** [review]

File: docs/learn/learn-staking.md (hunk 4 above)
Comment (salmad3) [reply docs/learn/learn-staking.md]: I had read up on the research wiki: https://research.web3.foundation/en/latest/polkadot/slashing/amounts.html#babe-equivocation which is where I landed at `0.05 * min((3 * (x - 1))) / n, 1)`. 

```
	fn slash_fraction(offenders: u32, validator_set_count: u32) -> Perbill {
		// the formula is min((3 * (k - (n / 10 + 1))) / n, 1) * 0.07
		// basically, 10% can be offline with no slash, but after that, it linearly climbs up to 7%
		// when 13/30 are offline (around 5% when 1/3 are offline).
		if let Some(threshold) = offenders.checked_sub(validator_set_count / 10 + 1) {
			let x = Perbill::from_rational(3 * threshold, validator_set_count);
			x.saturating_mul(Perbill::from_percent(7))
		} else {
			Perbill::default()
		}
	}
}
```

According to the source code, the previous was correct:
- The implementation here is that 10% can be offline and accounts for `(n / 10 + 1)))`.
- The parameter of 5% can be adjusted and is 7% in this case, but I am not sure why 7%. 
- Also true min(0 , 1) is 0 and makes sense as the fraction is zero for **isolated** cases.

In this case, should we add an example of 1 or 2 validators being offline, (under 10%) and an example for when more than 10% is offline, and when one third are offline? @DrW3RK

Comment (profradha) [reply docs/learn/learn-staking.md]: _The parameter of 5% can be adjusted and is 7% in this case, but I am not sure why 7%._

You are not adjusting anything. That is the output of that equation. The maximum anyone can get slashed is 7%. How? if the value is anything greater than 1, then min(value, 1) will return only 1 and subsequently the slashing hits the ceiling of 0.07 (7%)

If you plug in k = n/3 in that equation (assuming you have n = 100 validators), you end up with a value 3 * (33 - (100/ 10 + 1))) / 100) = 3*(33 -11)/100 = 0.66*0.07 which is approximately 5%

Also,

The implementation here is that 10% can be offline and accounts for (n / 10 + 1))), which is subtracted from the total **offending** validators in the active set.

Comment (salmad3) [reply docs/learn/learn-staking.md]: Ah, thank you

File: docs/learn/learn-staking.md [hunk 5]
```diff
@@ -297,18 +299,24 @@ available [here](../maintain/maintain-guides-secure-validator.md).
...
+min((3 * (1 - (100 / 10 + 1))) / 100, 1) * 0.07 which comes out to no slash.
 
-If 14 validator are unresponsiveness, the unresponsiveness is > 10% and grows linearly by 7%. 
-The validators would be slashed at a fraction of min((3 * (14 - (100 / 10 + 1))) / 100, 1) * 0.07 which 
-comes out to around a 1% slash.
+> The first example takes into account a unresponsiveness of < 10%, resulting in no slash.
 
-If 60 validators are unresponsiveness, the unresponsiveness of validators is > 1/3 of the all 
+If 14 validator are unresponsiveness, the unresponsiveness is > 10%. The validators would be slashed 
+at a fraction of min((3 * (14 - (100 / 10 + 1))) / 100, 1) * 0.07 which comes out to around a 0.06%
```
Comment (profradha) [inline docs/learn/learn-staking.md]: min((3 * (14 - (100 / 10 + 1))) / 100, 1) * 0.07  comes out to be around 2% slashing (Correction: It is 0.6%)

**Review by DrW3RK — CHANGES_REQUESTED** [review]

**Comment by DrW3RK** [comment]
@DannyS03 Let me work on this PR to avoid further back and forth. I will add the content and assign it for your review.

File: docs/learn/learn-staking.md (hunk 5 above)
Comment (DannyS03) [reply docs/learn/learn-staking.md:306]: 1%

Comment (DannyS03) [reply docs/learn/learn-staking.md:306]: And isn't 3 * (33 - (100/ 10 + 1))) / 100) = 3*(33 -11)/100 = 0.66*0.07 supposed to be  3 * (33 - (100/ 10 + 1))) / 100) = 3*(33 -9.09)/100 = 0.7173*0.07

Comment (salmad3) [reply docs/learn/learn-staking.md]: we're both calculating wrong 

I used the 11 provided in the example you gave for 100 / 10 + 1

14 should be min((3 * (14 - (100 / 10 + 1))) / 100, 1) * 0.07

-> min((3 * (14 - 9.09)) / 100, 1) * 0.07 -> min((3 * 4.91) / 100, 1) * 0.07 -> min((3 * 4.91) / 100, 1) * 0.07 -> min(14.73) / 100, 1) * 0.07 -> min(0.1473, 1) * 0.07 -> 1%

Comment (profradha) [reply docs/learn/learn-staking.md]: Yeah, please see my new writeup

Comment (profradha) [reply docs/learn/learn-staking.md]: @DannyS03 The division takes precedence before addition

Comment (salmad3) [reply docs/learn/learn-staking.md]: right yes, sorry lol typo on my end I wrote it out like min((3 * (14 - (100 / (10 + 1))) / 100, 1) * 0.07

**Review by DrW3RK — APPROVED** [review]

**Comment by DrW3RK** [comment]
@DannyS03 review the changes made and merge the PR. I believe all the examples in the writeup are accurate.

**Pull request merged**